server/src/main/java/org/elasticsearch/plugins/CompositeClassLoader.java

```diff
@@ -47,10 +47,14 @@ protected Class<?> findClass(String name) throws ClassNotFoundException {
         throw new ClassNotFoundException(name);
     }
 
-    /**
-     * Return a new classloader across the parent and extended loaders.
-     */
-    public static ExtendedPluginsClassLoader create(ClassLoader parent, List<ClassLoader> extendedLoaders) {
-        return new ExtendedPluginsClassLoader(parent, extendedLoaders);
+    @Override
+    public Enumeration<URL> getResources(String name) throws IOException {
+        List<URL> urls = new ArrayList<>();
+        super.getResources(name).asIterator().forEachRemaining(urls::add);
+        for (var loader : loaders) {
+            Enumeration<URL> resources = loader.getResources(name);
+            resources.asIterator().forEachRemaining(urls::add);
```

Comment:
Shouldn't we break the `loaders` iteration if the resource is loaded?

Reply:
No, this should return a enumeration of URLs over all loaders, not just the first.

What might be worth doing is using a set to de-duplicate entries.